uni/ulsp/folding.icn

```diff
@@ -0,0 +1,120 @@
+package ulsp
+
+link regexp
+link ximage
+
+#  Processes all folding ranges in a document and constructs a result to return to the client.
+class FoldingRangeHandler(filehandler)
+
+   #  Sets the class variables for the FoldingRangeHandler.
+   #  Params: new_filehandler - The filehandler object created from the parameters of the
+   #                             request sent by the client.
+   method setVariables(new_filehandler)
+      filehandler := new_filehandler
+   end
+
+   #  Searches through the provided filehandler for all internal positions of symbols and
+   #  constructs a result to send back to the client for all folding ranges.
+   method run()
+      local results_table, tokenKind, tokenName, entry, startLine, endLine, result
+
+      results_table := []
+
+      every tokenKind := key(filehandler.internal_positions) do {
+         every tokenName := key(filehandler.internal_positions[tokenKind]) do {
+            entry := filehandler.internal_positions[tokenKind][tokenName]
+
+            if member(entry, "range") then {
+               startLine := entry["range"]["start"][1]
+               endLine := entry["range"]["end"][1] - 1
+
+               if tokenKind == "initially" then endLine+:=1
+               if startLine < endLine then {
+                  put(results_table, table(
+                     "startLine", startLine,
+                     "endLine", endLine
+                  ))
+               }
+            }
+         }
+      }
+      every result := !findFoldingRanges() do put(results_table, result)
```

Comment:
you can eliminate the loop variable: 
```
every put(results_table,  !findFoldingRanges())
```

Reply:
Looking again at `findFoldingRanges()`, looks like it returns a list. In this case you just want to concatenate the two lists:
```
results_table  := result_table ||| findFoldingRanges()
```
or just
```
results_table  |||:= findFoldingRanges()
```